src/lib/fountain/fountain_encoder_stream.h

```diff
@@ -26,20 +30,28 @@ class fountain_encoder_stream
 	}
 
 public:
+
 	template <typename STREAM>
-	static fountain_encoder_stream create(STREAM& stream, unsigned buffer_size, uint8_t encode_id=0)
+	static fountain_encoder_stream::ptr create(STREAM& stream, unsigned buffer_size, uint8_t encode_id=0)
 	{
 		std::stringstream buffs;
 		if (stream)
-			 buffs << stream.rdbuf();
-		return fountain_encoder_stream(buffs.str(), buffer_size, encode_id);
+			buffs << stream.rdbuf();
+		return fountain_encoder_stream::ptr( new fountain_encoder_stream(buffs.str(), buffer_size, encode_id) );
```

Comment:
Ideally this would be a constructor -- alas, the template parameter disqualifies that. The `shared_ptr` fits the use cases the best, hopefully.